### PR TITLE
Migrate `pkg/kubelet/lifecycle,oom` to structured logging

### PR DESCRIPTION
**Pull request by mengjiao-liu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
For Structured Logging Migration: kubernetes/enhancements#1602
Migrate `pkg/kubelet/lifecycle,oom` to structured logging

#### Which issue(s) this PR fixes:

Ref #98976

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/lifecycle/handlers.go [hunk 1]
```diff
@@ -64,20 +64,20 @@ func (hr *handlerRunner) Run(containerID kubecontainer.ContainerID, pod *v1.Pod,
 		output, err := hr.commandRunner.RunInContainer(containerID, handler.Exec.Command, 0)
 		if err != nil {
 			msg = fmt.Sprintf("Exec lifecycle hook (%v) for Container %q in Pod %q failed - error: %v, message: %q", handler.Exec.Command, container.Name, format.Pod(pod), err, string(output))
-			klog.V(1).Infof(msg)
+			klog.V(1).InfoS("Exec lifecycle hook for Container in Pod failed", "execCommand", handler.Exec.Command, "containerName", container.Name, "pod", klog.KObj(pod), "err", err, "message", string(output))
```
Comment (yuzhiquan) [inline pkg/kubelet/lifecycle/handlers.go:67]: how about `Errors` method?

File: pkg/kubelet/lifecycle/handlers.go [hunk 2]
```diff
@@ -64,20 +64,20 @@ func (hr *handlerRunner) Run(containerID kubecontainer.ContainerID, pod *v1.Pod,
...
 		}
 		return msg, err
 	case handler.HTTPGet != nil:
 		msg, err := hr.runHTTPHandler(pod, container, handler)
 		if err != nil {
 			msg = fmt.Sprintf("Http lifecycle hook (%s) for Container %q in Pod %q failed - error: %v, message: %q", handler.HTTPGet.Path, container.Name, format.Pod(pod), err, msg)
-			klog.V(1).Infof(msg)
+			klog.V(1).InfoS("Http lifecycle hook for Container in Pod failed", "httpPath", handler.HTTPGet.Path, "containerName", container.Name, "pod", klog.KObj(pod), "err", err, "message", msg)
```
Comment (yuzhiquan) [inline pkg/kubelet/lifecycle/handlers.go:74]: how about `Errors` method?

File: pkg/kubelet/lifecycle/handlers.go [hunk 3]
```diff
@@ -110,7 +110,7 @@ func (hr *handlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 	if len(host) == 0 {
 		status, err := hr.containerManager.GetPodStatus(pod.UID, pod.Name, pod.Namespace)
 		if err != nil {
-			klog.Errorf("Unable to get pod info, event handlers may be invalid.")
+			klog.ErrorS(nil, "Unable to get pod info, event handlers may be invalid.")
```
Comment (yuzhiquan) [inline pkg/kubelet/lifecycle/handlers.go:113]: klog.ErrorS(err, "Unable to get pod info, event handlers may be invalid.")

File: pkg/kubelet/lifecycle/handlers.go (hunk 1 above)
Comment (mengjiao-liu) [reply pkg/kubelet/lifecycle/handlers.go:67]: I didn't change the error level and followed the original implementation

File: pkg/kubelet/oom/oom_watcher_linux.go [hunk 4]
```diff
@@ -73,15 +73,15 @@ func (ow *realWatcher) Start(ref *v1.ObjectReference) error {
...
+				klog.V(1).InfoS("Got sys oom event", "event", event)
 				eventMsg := "System OOM encountered"
 				if event.ProcessName != "" && event.Pid != 0 {
 					eventMsg = fmt.Sprintf("%s, victim process: %s, pid: %d", eventMsg, event.ProcessName, event.Pid)
 				}
 				ow.recorder.Eventf(ref, v1.EventTypeWarning, systemOOMEvent, eventMsg)
 			}
 		}
-		klog.Errorf("Unexpectedly stopped receiving OOM notifications")
+		klog.ErrorS(nil, "Unexpectedly stopped receiving OOM notifications")
```
Comment (yuzhiquan) [inline pkg/kubelet/oom/oom_watcher_linux.go:84]: If err is nil, may be use InfoS, or
```
klog.Errors(fmt.Errorf("Unexpectedly stopped receiving OOM notifications"), "")
```

File: pkg/kubelet/lifecycle/handlers.go (hunk 1 above)
Comment (yuzhiquan) [reply pkg/kubelet/lifecycle/handlers.go:67]: > I didn't change the error level and followed the original implementation

Fine, but with `"err"， err` set, may be ErrorS more available?

File: pkg/kubelet/lifecycle/handlers.go (hunk 3 above)
Comment (mengjiao-liu) [reply pkg/kubelet/lifecycle/handlers.go:113]: Maybe it's better to print error info. I'll take your advice

File: pkg/kubelet/lifecycle/handlers.go (hunk 1 above)
Comment (mengjiao-liu) [reply pkg/kubelet/lifecycle/handlers.go:67]: yes, I will fix it

File: pkg/kubelet/lifecycle/handlers.go (hunk 2 above)
Comment (mengjiao-liu) [reply pkg/kubelet/lifecycle/handlers.go:74]: Thanks, I will fix it

File: pkg/kubelet/oom/oom_watcher_linux.go [hunk 5]
```diff
@@ -73,15 +73,15 @@ func (ow *realWatcher) Start(ref *v1.ObjectReference) error {
...
+				klog.V(1).InfoS("Got sys oom event", "event", event)
 				eventMsg := "System OOM encountered"
 				if event.ProcessName != "" && event.Pid != 0 {
 					eventMsg = fmt.Sprintf("%s, victim process: %s, pid: %d", eventMsg, event.ProcessName, event.Pid)
 				}
 				ow.recorder.Eventf(ref, v1.EventTypeWarning, systemOOMEvent, eventMsg)
 			}
 		}
-		klog.Errorf("Unexpectedly stopped receiving OOM notifications")
+		klog.ErrorS(fmt.Errorf("Unexpectedly stopped receiving OOM notifications"), "")
```
Comment (ehashman) [inline pkg/kubelet/oom/oom_watcher_linux.go:84]:
```suggestion
		klog.ErrorS(nil, "Unexpectedly stopped receiving OOM notifications")
```

**Review by ehashman — CHANGES_REQUESTED** [review]
/lgtm cancel

caught one small issue

**Comment by mengjiao-liu** [comment]
> /lgtm cancel
> 
> caught one small issue

I've fixed this and rebase commit

File: pkg/kubelet/oom/oom_watcher_linux.go (hunk 5 above)
Comment (yuzhiquan) [reply pkg/kubelet/oom/oom_watcher_linux.go]: @ehashman if err is nil, may be InfoS more available?

File: pkg/kubelet/lifecycle/handlers.go [hunk 6]
```diff
@@ -64,20 +64,20 @@ func (hr *handlerRunner) Run(containerID kubecontainer.ContainerID, pod *v1.Pod,
...
-			klog.V(1).Infof(msg)
+			klog.V(1).ErrorS(err, "Exec lifecycle hook for Container in Pod failed", "execCommand", handler.Exec.Command, "containerName", container.Name, "pod", klog.KObj(pod), "message", string(output))
 		}
 		return msg, err
 	case handler.HTTPGet != nil:
 		msg, err := hr.runHTTPHandler(pod, container, handler)
 		if err != nil {
 			msg = fmt.Sprintf("Http lifecycle hook (%s) for Container %q in Pod %q failed - error: %v, message: %q", handler.HTTPGet.Path, container.Name, format.Pod(pod), err, msg)
-			klog.V(1).Infof(msg)
+			klog.V(1).ErrorS(err, "Http lifecycle hook for Container in Pod failed", "httpPath", handler.HTTPGet.Path, "containerName", container.Name, "pod", klog.KObj(pod), "message", msg)
```
Comment (ehashman) [inline pkg/kubelet/lifecycle/handlers.go:74]:
```suggestion
			klog.V(1).ErrorS(err, "HTTP lifecycle hook for Container in Pod failed", "path", handler.HTTPGet.Path, "containerName", container.Name, "pod", klog.KObj(pod))
```

We can drop `msg`, as it is a copy of this string.

File: pkg/kubelet/lifecycle/predicate.go [hunk 7]
```diff
@@ -98,7 +98,7 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 	fit := len(reasons) == 0 && err == nil
 	if err != nil {
 		message := fmt.Sprintf("GeneralPredicates failed due to %v, which is unexpected.", err)
```
Comment (ehashman) [inline pkg/kubelet/lifecycle/predicate.go:100]:
```suggestion
```

File: pkg/kubelet/lifecycle/predicate.go [hunk 8]
```diff
@@ -98,7 +98,7 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
...
-		klog.Warningf("Failed to admit pod %v - %s", format.Pod(admitPod), message)
+		klog.InfoS("Failed to admit pod", "pod", klog.KObj(admitPod), "message", message)
```
Comment (ehashman) [inline pkg/kubelet/lifecycle/predicate.go:101]:
```suggestion
		klog.InfoS("Failed to admit pod, GeneralPredicates failed", "pod", klog.KObj(admitPod), "err", err)
```

File: pkg/kubelet/lifecycle/predicate.go [hunk 9]
```diff
@@ -110,7 +110,7 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 		fit = len(reasons) == 0 && err == nil
 		if err != nil {
 			message := fmt.Sprintf("Unexpected error while attempting to recover from admission failure: %v", err)
```
Comment (ehashman) [inline pkg/kubelet/lifecycle/predicate.go:112]:
```suggestion
```

File: pkg/kubelet/lifecycle/predicate.go [hunk 10]
```diff
@@ -110,7 +110,7 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
...
-			klog.Warningf("Failed to admit pod %v - %s", format.Pod(admitPod), message)
+			klog.InfoS("Failed to admit pod", "pod", klog.KObj(admitPod), "message", message)
```
Comment (ehashman) [inline pkg/kubelet/lifecycle/predicate.go:113]:
```suggestion
			klog.InfoS("Failed to admit pod, unexpected error while attempting to recover from admission failure", "pod", klog.KObj(admitPod), "err", err)
```

File: pkg/kubelet/lifecycle/predicate.go [hunk 11]
```diff
@@ -123,7 +123,7 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 		var message string
 		if len(reasons) == 0 {
 			message = fmt.Sprint("GeneralPredicates failed due to unknown reason, which is unexpected.")
```
Comment (ehashman) [inline pkg/kubelet/lifecycle/predicate.go:125]:
```suggestion
```

File: pkg/kubelet/lifecycle/predicate.go [hunk 12]
```diff
@@ -123,7 +123,7 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
...
-			klog.Warningf("Failed to admit pod %v - %s", format.Pod(admitPod), message)
+			klog.InfoS("Failed to admit pod", "pod", klog.KObj(admitPod), "message", message)
```
Comment (ehashman) [inline pkg/kubelet/lifecycle/predicate.go:126]:
```suggestion
			klog.InfoS("Failed to admit pod: GeneralPredicates failed due to unknown reason, which is unexpected", "pod", klog.KObj(admitPod))
```

File: pkg/kubelet/lifecycle/predicate.go [hunk 13]
```diff
@@ -136,15 +136,15 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 		case *PredicateFailureError:
 			reason = re.PredicateName
 			message = re.Error()
-			klog.V(2).Infof("Predicate failed on Pod: %v, for reason: %v", format.Pod(admitPod), message)
+			klog.V(2).InfoS("Predicate failed on Pod", "pod", format.Pod(admitPod), "reason", message)
```
Comment (ehashman) [inline pkg/kubelet/lifecycle/predicate.go:139]:
```suggestion
			klog.V(2).InfoS("Predicate failed on Pod", "pod", format.Pod(admitPod), "err", message)
```

File: pkg/kubelet/lifecycle/predicate.go [hunk 14]
```diff
@@ -136,15 +136,15 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
...
 		case *InsufficientResourceError:
 			reason = fmt.Sprintf("OutOf%s", re.ResourceName)
 			message = re.Error()
-			klog.V(2).Infof("Predicate failed on Pod: %v, for reason: %v", format.Pod(admitPod), message)
+			klog.V(2).InfoS("Predicate failed on Pod", "pod", format.Pod(admitPod), "reason", message)
```
Comment (ehashman) [inline pkg/kubelet/lifecycle/predicate.go:143]:
```suggestion
			klog.V(2).InfoS("Predicate failed on Pod", "pod", format.Pod(admitPod), "err", message)
```

File: pkg/kubelet/lifecycle/predicate.go [hunk 15]
```diff
@@ -136,15 +136,15 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
...
 		default:
 			reason = "UnexpectedPredicateFailureType"
 			message = fmt.Sprintf("GeneralPredicates failed due to %v, which is unexpected.", r)
-			klog.Warningf("Failed to admit pod %v - %s", format.Pod(admitPod), message)
+			klog.InfoS("Failed to admit pod", "pod", klog.KObj(admitPod), "message", message)
```
Comment (ehashman) [inline pkg/kubelet/lifecycle/predicate.go:147]:
```suggestion
			klog.InfoS("Failed to admit pod", "pod", klog.KObj(admitPod), "err", message)
```

**Review by ehashman — COMMENTED** [review]
/lgtm cancel
/cc @serathius

**Comment by ehashman** [comment]
Sorry for the back and forth, I think this is my first full review of this CL!

File: pkg/kubelet/lifecycle/predicate.go (hunk 7 above)
Comment (mengjiao-liu) [reply pkg/kubelet/lifecycle/predicate.go:100]: But the `msg`  variable is used [below](https://github.com/kubernetes/kubernetes/blob/9735ef75a62e19165b1a9c814b42789548729457/pkg/kubelet/lifecycle/predicate.go#L117)
```
		return PodAdmitResult{
			Admit:   fit,
			Reason:  "UnexpectedAdmissionError",
			Message: message,
		}
```

File: pkg/kubelet/lifecycle/predicate.go (hunk 9 above)
Comment (mengjiao-liu) [reply pkg/kubelet/lifecycle/predicate.go:112]: Same as above

File: pkg/kubelet/lifecycle/predicate.go (hunk 11 above)
Comment (mengjiao-liu) [reply pkg/kubelet/lifecycle/predicate.go:125]: Same as above

File: pkg/kubelet/lifecycle/handlers.go (hunk 6 above)
Comment (mengjiao-liu) [reply pkg/kubelet/lifecycle/handlers.go:74]: Thanks for  your review and I will change it

File: pkg/kubelet/lifecycle/predicate.go (hunk 13 above)
Comment (mengjiao-liu) [reply pkg/kubelet/lifecycle/predicate.go:139]: Thanks for  your review and I will change it

File: pkg/kubelet/lifecycle/predicate.go (hunk 14 above)
Comment (mengjiao-liu) [reply pkg/kubelet/lifecycle/predicate.go:143]: Thanks for  your review and I will change it

File: pkg/kubelet/lifecycle/predicate.go (hunk 15 above)
Comment (mengjiao-liu) [reply pkg/kubelet/lifecycle/predicate.go:147]: Thank,  I will change it

**Review by kikisdeliveryservice — COMMENTED** [review]
All requested changes made (excluding message deletion which is used later in file), author also updated all logs in selected files, changes look good.

/lgtm

**Pull request merged**